test/Microsoft.AspNet.Mvc.Razor.Test/Compilation/ExpressionRewriterTest.cs

```diff
@@ -19,6 +19,37 @@ namespace Microsoft.AspNet.Mvc.Razor.Compilation
     public class ExpressionRewriterTest
     {
         [Fact]
+        public void ExpressionRewriter_DoesNotThrowsOnUnknownTypes()
+        {
+            // Arrange
+            var source = @"
+using System.Threading.Tasks;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Mvc.Razor;
+
+public class ExamplePage : RazorPage
+{
+    public IViewComponentHelper Component { get; set; }
+
+    public override async Task ExecuteAsync()
+    {
+        Write(
+            await Component.InvokeAsync(
+                ""SomeComponent"",
+                item => new HelperResult((__razor_template_writer) => WriteLiteralTo(__razor_template_writer, ""Hello World""))));
+        }
+    }
+";
+            var tree = CSharpSyntaxTree.ParseText(source);
+            var compilation = Compile(tree, allowErrors: true);
```

Comment:
Ryan and I poked around this yesterday and it's truly an anomaly. When attempting to replicate the user who reported this bugs scenario by fabricating a near similar example you end up running into compilation errors prior to ever hitting the `ExpressionRewriter`. However, attempting the users exact code side by side failed after compile. 

After a significant time investment into trying to determine why Roslyn was parsing/compiling the way it was for the users code but not ours (we tried to duplicate without `ViewComponents`) we threw in the towel and decided making the `ExpressionRewriter` more resilient was an appropriate fix.

Can add a comment explaining this background.